R/fit.R

```diff
@@ -391,16 +391,20 @@ mmrm <- function(formula,
   if (control$method %in% c("Kenward-Roger", "Kenward-Roger-Linear") && !reml) {
     stop("Kenward-Roger only works for REML")
   }
+  covariance <- h_reconcile_cov_struct(formula, covariance)
+  formula_parts <- h_mmrm_tmb_formula_parts(formula, covariance)
 
-  attr(data, which = "dataname") <- toString(match.call()$data)
+  if (!missing(data)) {
+    attr(data, which = "dataname") <- toString(match.call()$data)
+  } else {
+    data <- model.frame(formula_parts$full_formula)
```

Comment:
please note that the environment of the formula matters. So here in `h_mmrm_tmb_formula_parts` we are not creating new formulas but updating the original formula is good: the environment is not affected